src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -33,6 +35,362 @@ def setUp(self):
+          self._sas_admin.InjectEscSensorDataRecord({'record': esc_sensor})
+      # step 7
+      # Notify the SAS UUT about the SAS Test Harness
+      sas_th_config = config['sasTestHarnessConfig']		
+      certificate_hash = getCertificateFingerprint(sas_th_config['serverCert'])
+      self._sas_admin.InjectPeerSas({'certificateHash': certificate_hash,
+                                  'url': sas_th_config['url']})
+
+      response = self.TriggerFullActivityDumpAndWaitUntilComplete(sas_th_config['serverCert'], sas_th_config['serverKey'])
+      # check dump message format
+      self.assertContainsRequiredFields("FullActivityDump.schema.json", response)
+      # an array for each record type
+      cbsd_dump_data = []
+      ppa_dump_data = []
+      esc_sensor_dump_data = []
+        
+      # step 8 and check   
+      # download dump files and fill corresponding arrays
+      for dump_file in response['files']:
+          self.assertContainsRequiredFields("ActivityDumpFile.schema.json",
+                                              dump_file)
+          downloaded_file = None
+          if dump_file['recordType'] != 'CoordinationEvent':                
+              downloaded_file = self._sas.DownloadFile(dump_file['url'],\
+				  sas_th_config['serverCert'], sas_th_config['serverKey'])
+          if dump_file['recordType'] ==  'cbsd':
+              cbsd_dump_data.extend(downloaded_file['recordData'])   
+          elif dump_file['recordType'] ==  'esc_sensor':
+              esc_sensor_dump_data.extend(downloaded_file['recordData'])
+          elif dump_file['recordType'] ==  'zone':
+              ppa_dump_data.extend(downloaded_file['recordData'])
+          else:
+              self.assertEqual('CoordinationEvent', dump_file['recordType'])
+        
+      # verify the length of records equal to the inserted ones
+      self.assertEqual(len(config['registrationRequests']), len(cbsd_dump_data))
+      self.assertEqual(len(config['ppaRecords']), len(ppa_dump_data))
+      self.assertEqual(len(config['escSensorRecords']), len(esc_sensor_dump_data))
+      # verify the schema of record and first two parts of PPA record Id  
+      for ppa_record in ppa_dump_data:
+        self.assertContainsRequiredFields("ZoneData.schema.json", ppa_record)              
+        self.assertEqual(ppa_record['id'].split("/")[0], 'zone')
+        self.assertEqual(ppa_record['id'].split("/")[1], 'ppa')
+        self.assertEqual(ppa_record['id'].split("/")[2], self._sas._sas_admin_id)
+        del ppa_record['id']
+        # TODO : to update the following check according when PAL grants are supported in this test case
+        if 'cbsdReferenceId' in ppa_record['ppaInfo']:
+          self.assertFalse(ppa_record['ppaInfo']['cbsdReferenceId'])
+          del ppa_record['ppaInfo']['cbsdReferenceId']
+        # verify that the injected ppas exist in the dump files
+        # TODO: add Check for GeoJSON Winding of ppa polygon  
+        exist_in_dump = False
```

Comment:
If needed, you can now add here the winding check using the new routines

Reply:
I don't know if we need to do the winding check if ,as you mentioned before, it is not recommended for GeoJSON?